ReSharper.FSharp/src/FSharp.Psi.Daemon/src/Analyzers/LambdaAnalyzer.fs

```diff
@@ -85,7 +86,41 @@ type LambdaAnalyzer() =
 
         compareArgsRec expr 0 null
 
+    let hasExplicitConversion (lambda: ILambdaExpr) =
+        let expr = lambda.Expression.IgnoreInnerParens()
+
+        let lambdaType = lambda.TryGetFcsType()
+        if isNull lambdaType then false else
+
+        let lambdaParamsCount = lambda.Patterns.Count
+        let mutable lambdaReturnType = lambdaType
+        let mutable i = lambdaParamsCount
+
+        while i > 0 && lambdaReturnType.IsFunctionType do
+            lambdaReturnType <- lambdaReturnType.GenericArguments[1]
+            i <- i - 1
+
+        let exprType =
+            match expr with
+            | :? IPrefixAppExpr as prefixApp ->
+                match prefixApp.FunctionExpression with
+                | :? IReferenceExpr as reference ->
+                    match reference.Reference.GetFcsSymbol() with
```

Comment:
It's often much easier to read psi-related code later when expressions are called expressions, and references are called references, not when both share `reference` name. 🙂 
```suggestion
                | :? IReferenceExpr as referenceExpr ->
                    match referenceExpr.Reference.GetFcsSymbol() with
```